src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -185,15 +196,67 @@ private void closeExecutorService() {
+        String type = clazz.getSimpleName();
+        return metricName(name, type, group, tags);
+    }
+
     static MetricName metricName(Class<?> clazz, String name) {
         String group = clazz.getPackageName();
         String type = clazz.getSimpleName();
         return metricName(name, type, group);
     }
 
-    static MetricName metricName(String name, String type, String group) {
-        String mBeanName = String.format("%s:type=%s,name=%s", group, type, name);
-        return new MetricName(group, type, name, SCOPE, mBeanName);
+    /**
+     * Generate a Yammer metric name
+     *
+     * @param name  the name of the Metric.
+     * @param type  the type to which the Metric belongs.
+     * @param group the group to which the Metric belongs type
+     * @return the MetricName object derived from the arguments.
+     */
+    public static MetricName metricName(String name, String type, String group) {
+        final String sanitisedGroup = sanitise(group);
+        final String sanitisedType = sanitise(type);
+        final String sanitisedName = sanitise(name);
+        String mBeanName = String.format("%s:type=%s,name=%s", sanitisedGroup, sanitisedType, sanitisedName);
+        return new MetricName(sanitisedGroup, sanitisedType, sanitisedName, SCOPE, mBeanName);
+    }
+
+    /**
+     * Generate a Yammer metric name
+     *
+     * @param name  the name of the Metric.
+     * @param type  the type to which the Metric belongs.
+     * @param group the group to which the Metric belongs type
+     * @param tags  an ordered set of key value mappings
+     * @return the MetricName object derived from the arguments.
+     */
+    public static MetricName metricName(String name, String type, String group, LinkedHashMap<String, String> tags) {
+        final String tagValues = tags.entrySet().stream().map(entry -> String.format("%s=%s", sanitise(entry.getKey()), sanitise(entry.getValue()))).collect(Collectors.joining(","));
+        String mBeanName;
+        final String sanitisedGroup = sanitise(group);
+        final String sanitisedType = sanitise(type);
+        final String sanitisedName = sanitise(name);
+        if (!tagValues.isBlank()) {
+            mBeanName = String.format("%s:type=%s,name=%s,%s", sanitisedGroup, sanitisedType, sanitisedName, tagValues);
+        } else {
+            mBeanName = String.format("%s:type=%s,name=%s", sanitisedGroup, sanitisedType, sanitisedName);
+        }
+        return new MetricName(sanitisedGroup, sanitisedType, sanitisedName, SCOPE, mBeanName);
+    }
+
+    private static String sanitise(String name) {
+        // It would be more efficient to create a single pattern and replace everything once.
+        // However, I think this makes things clearer and easier to understand.
+        return name.replaceAll(":", "")
+                .replaceAll("\\?", "")
+                .replaceAll("\\*", "")
+                .replaceAll("//", "")
+                .replaceAll("\\$$", "")
```

Comment:
I think its because `$` is used as a marker to denote inner classes so its only illegal if its a trailing character. 

I will add a comment to that effect.

Reply:
done in b9c43213a1077c807fa8dc300844f8a5c225eaae